### ISSUE TITLE
Pyright reports incompatible types in some circumstances when inheriting from a class implementing `Generic`

### ISSUE THREAD

**Issue by kennipj** [microsoft/pyright#1796]
Pyright reports incompatible types when subclasses inheriting from a class with generic types specify itself as the type, when the superclass also inherits from other classes. Easier explained with code:

```py
from typing import Generic, TypeVar
T = TypeVar("T", bound="A")


class A:
    pass


class B(A, Generic[T]):
    some_field: T


class C(B["C"]):
    # Type "C" cannot be assigned to type variable "T@B"
    #  Type "C" is incompatible with bound type "A" for type variable "T@B"
    #    "C" is incompatible with "A"
    pass
```

However, if I remove the inheritance of `A` from `B`, and add it to `C`, then there are no issues:

```py
from typing import Generic, TypeVar
T = TypeVar("T", bound="A")


class A:
    pass


class B(Generic[T]):
    some_field: T


class C(A, B["C"]):
    pass
```

This is quite impractical, as I have a use case where I'd like to specify this inheritance in `B`, so that any subclasses of `B` don't all need to explicitly inherit from `A`. 

Mypy does not report any issues with the original example.

Tested on version `1.1.135`.

**Comment by erictraut** [comment]
Thanks for the bug report. This will be fixed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.136, which I just published. It will also be included in the next release of pylance.
